components/overledger/actions/sign-a-transaction/sign-a-transaction.mjs

```diff
@@ -0,0 +1,77 @@
+import overledger from "../../overledger.app.mjs";
+import { UNIT_OPTIONS } from "../../common/constants.mjs";
+
+export default {
+  key: "overledger-sign-a-transaction",
+  name: "Sign a transaction",
+  description: "Sign a transaction using Overledger",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    overledger,
+    environment: {
+      propDefinition: [
+        overledger,
+        "environment",
+      ],
+    },
+    locationTechnology: {
+      type: "string",
+      label: "Location Technology",
+      description: "The blockchain technology used for this transaction, e.g., ethereum, substrate - required in order to set the dltfee",
+      //previous prep step output - reuired in this step in order to set the dltfee unit
+      default: ({ steps }) => steps.prepare_smart_contract_transaction?.locationTechnology || "ethereum",
+    },
+    keyId: {
+      type: "string",
+      label: "Signing Account ID",
+      description: "The blockchain account that will sign the transaction.",
+    },
+    requestId: {
+      type: "string",
+      label: "requestId",
+      description: "The ID assigned to a preparation request in Overledger. This should be set to the requestId parameter found in the response object of the 'Prepare Smart Contract Transaction' Overledger action.",
+    },
+    transactionSigningResponderName: {
+      type: "string",
+      label: "Transaction Signing Responder Name",
+      description: "The name of the Transaction Signing Responder you would like to use. The CTA Transaction Signing Responder is the Quant-provided signer for testnet accounts.",
+    },
+    nativeData: {
+      type: "object",
+      label: "Native Data",
+      description: "A JSON object representing the transaction required to be signed.",
+      // Reference the output of the previous step
+      default: ({ steps }) => steps.prepare_smart_contract_transaction?.nativeData || {},
```

Comment:
It doesn't work
```suggestion
```

Reply:
Hi @luancazarine . Do you suggest I just remove this default value? Has it been tested with the Prep smart contract  step before hand?

Reply:
Yes, just remove it. Props don't have access to steps, it will return error if you try to publish the component

Reply:
done